General/DeviceRegistration/DeviceRegistrationWorker.swift

```diff
@@ -82,9 +85,11 @@ class DeviceRegistrationWorker: DeviceRegistering {
             }
             .subscribe(onNext: { [self] value in
                 try? store.save(value, for: storeKey)
+                appGroupStore?.saveSDKReady(value)
                 deviceRegisteredSubject.send(value)
                 completion(value)
             }, onError: { [self] _ in
+                appGroupStore?.saveSDKReady(false)
```

Comment:
Is there any scenario where we have registered the device, and then we call register device again and it fails, meaning SDKReady would be false?

Reply:
yes however if the register device call fails for some reason, app has retry mechanism to make the call again until it success.